From 73e2d045a14e9fcd42efa7a194da9b3ffb58f91a Mon Sep 17 00:00:00 2001
From: Tatsuo ONISHI <onishi@latmos.ipsl.fr>
Date: Wed, 12 Oct 2022 09:53:18 +0200
Subject: [PATCH 20/70] (Commit update) Adaptations of the previous commit to
 v4.4.1

Adaptation of 'Enhancement I : Direct radiative effects of aerosols
 and ozone Enhancement II: Aerosol optical properties without effects of BC,
 SO4, NO3, NH4 or OC'

In the previous commit, these two files were excluded from the original commit.

modified files
--------------
phys/module_ra_rrtmg_sw.F
  * change all if(o3input.eq.2) conditions to if(o3input.eq.2.or.o3input.eq.3)

phys/module_ra_rrtmg_sw.F
  * change all if(o3input.eq.2) conditions to if(o3input.eq.2.or.o3input.eq.3)
  * change if(o3input.ne.2) condition to if(o3input.ne.2.and.o3input.ne.3)
---
 phys/module_ra_rrtmg_lw.F | 8 ++++++--
 phys/module_ra_rrtmg_sw.F | 8 ++++++--
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/phys/module_ra_rrtmg_lw.F b/phys/module_ra_rrtmg_lw.F
index e31d8c3..911d75f 100644
--- a/phys/module_ra_rrtmg_lw.F
+++ b/phys/module_ra_rrtmg_lw.F
@@ -12400,7 +12400,9 @@ CONTAINS
          call inirad (o3mmr,plev,kts,nlay-1)
 
 ! Steven Cavallo: Changed to nlayers from kte+1
-        if(o3input.eq.2) then
+!lm      if(o3input .eq. 2)then
+        if(o3input .eq. 2 .or. o3input .eq. 3) then
+!lm
          do k = kts, nlayers
             o3vmr(ncol,k) = o3mmr(k) * amdo
             if(k.le.kte)then
@@ -12419,7 +12421,9 @@ CONTAINS
         endif
 
 ! output o3 for o3input=0
-        IF (o3input.ne.2) THEN
+!TO2        IF (o3input.ne.2) THEN
+        IF (o3input.ne.2.and.o3input.ne.3) THEN
+!TO2
             DO K=kts,kte
                O33D(I,K,J)=O31D(K)
             ENDDO
diff --git a/phys/module_ra_rrtmg_sw.F b/phys/module_ra_rrtmg_sw.F
index c0eb328..2d8a841 100644
--- a/phys/module_ra_rrtmg_sw.F
+++ b/phys/module_ra_rrtmg_sw.F
@@ -10597,7 +10597,9 @@ CONTAINS
             QV1D(K)=max(0.,QV1D(K))
          ENDDO
 
-         IF (o3input.eq.2) THEN
+!lm        if(o3input .eq. 2)then
+        if(o3input .eq. 2 .or. o3input .eq. 3) then
+!lm
             DO K=kts,kte
                O31D(K)=O33D(I,K,J)
             ENDDO
@@ -10918,7 +10920,9 @@ CONTAINS
 ! Get ozone profile including amount in extra layer above model top
          call inirad (o3mmr,plev,kts,kte)
 
-        if(o3input.eq.2) then
+!lm        if(o3input .eq. 2)then
+        if(o3input .eq. 2 .or. o3input .eq. 3) then
+!lm
          do k = kts, kte+1
             o3vmr(ncol,k) = o3mmr(k) * amdo
             if(k.le.kte)then
-- 
2.31.1

